pkg/gce-pd-csi-driver/controller.go

```diff
@@ -249,6 +264,11 @@ func (gceCS *GCEControllerServer) ControllerPublishVolume(ctx context.Context, r
 		return nil, status.Error(codes.NotFound, fmt.Sprintf("Could not find volume with ID %v: %v", volumeID, err))
 	}
 
+	if _, alreadyExists := gceCS.volumes.LoadOrStore(volKey.String(), true); alreadyExists {
```

Comment:
This gets a little tricky.  We actually want to serialize attach/detach operations on the same volume.  But we want to allow attach on the same volume in parallel if they're different nodes. (for rox access mode)

Reply:
Are you then worried about seralizating volume operations for other calls like CreateSnapshot or DeleteSnapshot? I added it here just in case but I can remove those.

Reply:
I think seralizing amongst (un)provision/attach/snapshot for the same volume is fine. Let's be safe for now, and we can considering relaxing it in the future if we find it's an issue. The only corner case is the attaching from ROX to multiple nodes. That should be allowed to happen in parallel.